UIBarButtonItem+Action.swift

```diff
@@ -0,0 +1,92 @@
+import UIKit
+import RxSwift
+import RxCocoa
+import ObjectiveC
+
+public extension UIBarButtonItem {
+	
+	/// Binds enabled state of action to button, and subscribes to rx_tap to execute action.
+	/// These subscriptions are managed in a private, inaccessible dispose bag. To cancel
+	/// them, set the rx_action to nil or another action.
+	public var rx_action: CocoaAction? {
+		get {
+			var action: CocoaAction?
+			doLocked {
+				action = objc_getAssociatedObject(self, &AssociatedKeys.Action) as? Action
+			}
+			return action
+		}
+		
+		set {
+			doLocked {
+				// Store new value.
+				objc_setAssociatedObject(self, &AssociatedKeys.Action, newValue, .OBJC_ASSOCIATION_RETAIN_NONATOMIC)
+				
+				// This effectively disposes of any existing subscriptions.
+				self.resetActionDisposeBag()
+				
+				// Set up new bindings, if applicable.
+				if let action = newValue {
+					action
+						.enabled
+						.bindTo(self.rx_enabled)
+						.addDisposableTo(self.actionDisposeBag)
+					
+					// Technically, this file is only included on tv/iOS platforms,
+					// so this optional will never be nil. But let's be safe 😉
+					let lookupControlEvent: ControlEvent<Void>?
+					
+					#if os(tvOS)
+						lookupControlEvent = self.rx_primaryAction
+					#elseif os(iOS)
+						lookupControlEvent = self.rx_tap
+					#endif
+					
+					guard let controlEvent = lookupControlEvent else {
+						return
+					}
+					
+					controlEvent
+						.subscribeNext { _ -> Void in
+							action.execute()
+						}
+						.addDisposableTo(self.actionDisposeBag)
+				}
+			}
+		}
+	}
+}
+
+// Note: Actions performed in this extension are _not_ locked
+// So be careful!
+private extension UIBarButtonItem {
+	private struct AssociatedKeys {
```

Comment:
I wonder if putting these in a shared superclass, like `UIControl`, might make sense. That way we wouldn't have to repeat this for `UIButton` too.

Actually, the whole `doLocked` and `actionDisoseBag` could be put there. Thoughts?